app/assets/javascripts/admin/controllers/admin-web-hooks-show.js.es6

```diff
@@ -0,0 +1,100 @@
+import { popupAjaxError } from 'discourse/lib/ajax-error';
+import computed from 'ember-addons/ember-computed-decorators';
+
+export default Ember.Controller.extend({
+  needs: ['adminWebHooks'],
+  eventTypes: Em.computed.alias('controllers.adminWebHooks.eventTypes'),
+  defaultEventTypes: Em.computed.alias('controllers.adminWebHooks.defaultEventTypes'),
+  contentTypes: Em.computed.alias('controllers.adminWebHooks.contentTypes'),
+
+  @computed('model.isSaving', 'saved', 'saveButtonDisabled')
+  savingStatus(isSaving, saved, saveButtonDisabled) {
+    if (isSaving) {
+      return I18n.t('saving');
+    } else if (!saveButtonDisabled && saved) {
+      return I18n.t('saved');
+    }
+    this.set('saved', false);
+    return '';
+  },
+
+  @computed('model.isNew')
+  saveButtonText(isNew) {
+    return isNew ? I18n.t('admin.web_hooks.create') : I18n.t('admin.web_hooks.save');
+  },
+
+  @computed('model.payload_url')
+  urlValidation(url) {
+    // If blank, fail without a reason
+    if (Ember.isEmpty(url)) {
+      return Discourse.InputValidation.create({
+        failed: true
+      });
+    }
+
+    if (!Discourse.Utilities.urlValid(url) || url.indexOf(' ') !== -1) {
```

Comment:
`url.indexOf(' ') !== -1` Can `Discourse.Utilities.urlValid(url)` catch this instead? Otherwise, we should trim the input